vendor/github.com/samuel/go-zookeeper/zk/dnshostprovider.go

```diff
@@ -0,0 +1,88 @@
+package zk
+
+import (
+	"fmt"
+	"net"
+	"sync"
+)
+
+// DNSHostProvider is the default HostProvider. It currently matches
+// the Java StaticHostProvider, resolving hosts from DNS once during
+// the call to Init.  It could be easily extended to re-query DNS
+// periodically or if there is trouble connecting.
+type DNSHostProvider struct {
```

Comment:
The only problem with the fail-stop model is that it prevents the cluster's current state from being served if only the ZKs fail. I don't know what we should prefer.